src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
```

Comment:
`_self` there is actually an instance variable, but I can see how the name is confusing :D I'll rename it to `_retainSelf`.

> but there better be timeout error callbacks on the finish, or calls GRPCCalls that wait forever will never get dealloc'ed

I'll be adding a way for the user to specify a timeout, but the plan was to have the default be infinite. Having a finite default value makes sense, but how to pick one?